matsim/src/main/java/org/matsim/core/population/routes/PopulationComparison.java

```diff
@@ -3,127 +3,185 @@
+                if (!isWithinDelta(plan1.getScore(), plan2.getScore(), delta)) {
+
+                    double maxScore = Double.NEGATIVE_INFINITY;
+                    for (Plan plan : person2.getPlans()) {
+                        if (plan.getScore() > maxScore) {
+                            maxScore = plan.getScore();
+                        }
+                    }
+
+                    log.warn("");
+                    log.warn("personId=" + person1.getId() + "; score1=" + plan1.getScore() + "; score2=" + plan2.getScore() + "; maxScore2=" + maxScore);
+                    log.warn("");
+
+                    result = Result.notEqual;
+
+                }
+            } else if (score1.isEmpty() && score2.isEmpty()) {
+            } else {
+                log.warn("");
+                log.warn(" selected plan scores not consistently present: p1: " + person1.getId() + " | p2: " + person2.getId());
+                result = Result.notEqual;
+            }
+
+            if (!equals(plan1.getPlanElements(), plan2.getPlanElements())) {
+                log.warn("");
+                log.warn(" selected plan elements not equal: p1: " + person1.getId() + " | p2: " + person2.getId());
+
+                for (PlanElement planElement : plan1.getPlanElements()) {
+                    log.warn(planElement);
+                }
+                log.warn("");
+                for (PlanElement planElement : plan2.getPlanElements()) {
+                    log.warn(planElement);
+                }
+                log.warn("");
+                result = Result.notEqual;
+            }
+        }
+        return result;
+    }
+
+    private static boolean isWithinDelta(double double1, double double2, double delta) {
+        return Math.abs(double1 - double2) < delta;
+    }
+
+    private static boolean isBothUndefinedOrWithinDelta(OptionalTime time1, OptionalTime time2, double delta) {
+        if (time1.isUndefined() ^ time2.isUndefined()) {
+            return false;
+        } else if(time1.isDefined()) {
+            return isWithinDelta((int) time1.seconds(), (int) time2.seconds(), delta);
+        }
+        return true;
+    }
+
+    public static boolean equals(List<PlanElement> planElements,
+                                 List<PlanElement> planElements2) {
+        return equals(planElements, planElements2, DEFAULT_DELTA);
+    }
+
+    public static boolean equals(List<PlanElement> planElements,
+                                 List<PlanElement> planElements2, double delta) {
+        int nElements = planElements.size();
+        if (nElements != planElements2.size()) {
+            return false;
+        } else {
+            for (int i = 0; i < nElements; i++) {
+                if (!equals(planElements.get(i), planElements2.get(i), delta)) {
+                    return false;
+                }
+            }
+        }
+        return true;
+    }
+
+    /* Warning: This is NOT claimed to be correct. (It isn't.)
```

Comment:
What are the known issues? Should we have a test to show it works for the existing use-case?

Reply:
this is a comment i copied over from a matsim test where this method used to live (the previous population comparison didnt test the plan elements)